### ISSUE TITLE
disappearing blocks

### ISSUE THREAD

**Issue by urushira** [msummers123/Movecraft-3#173]
I just ran into this issue tonight and am hoping you can provide insight. 
I've been building a too scale replica of the Graf Zepplin.  At 247 meters long and 34 meters high, it used 19728 iron blocks for its skin.  Because of its size, I needed to build the exterior in one spot before moving it to its permanent location. On the first rotation part of the tail broke off and stayed behind, then, in flight, over 2000 blocks simply disappeared leaving a large gaping hole where the skin used to be. I'm not sure I can actually repair it, but would like to understand the problem so i can avoid it in the future. I do get a lot of "Ship is too big to check its borders" errors when exiting the ships.. If you can provide me any insight into this issue or a work around, or a way to fix it, I would greatly appreciate it.. 
Thanks.
![2015-12-14_07 32 49](https://cloud.githubusercontent.com/assets/16305786/11811695/e3ecfdf2-a2eb-11e5-9400-6d8941d1ea25.png)
![2015-12-15_03 27 37](https://cloud.githubusercontent.com/assets/16305786/11811696/e3f0ee8a-a2eb-11e5-9520-9a62711e586c.png)
![2015-12-15_03 48 59](https://cloud.githubusercontent.com/assets/16305786/11811697/e3f28498-a2eb-11e5-8120-8fa6d687b4e1.png)
![2015-12-15_04 12 48](https://cloud.githubusercontent.com/assets/16305786/11811698/e3f6943e-a2eb-11e5-8feb-6b54206b791c.png)
![2015-12-15_04 16 57](https://cloud.githubusercontent.com/assets/16305786/11811699/e3fa1a00-a2eb-11e5-9796-74dc6d377f3b.png)

**Comment by mwkaicz** [comment]
was there that line, after crash, or you build it after a hole was created?

Ship is too big to check its borders ... do you have this warning with this concrete ship? really during exiting the ships?  It should warn you that your ship is connected to something, but not "Ship is too big to check its borders" ... this error should come only from detection task ... that's the part when you are activating the new ship.
... this is usual problem when your ship is connected to the land and materials on the land are also in your .craft configuration file ... so plugin doesn't know where to stop looking for parts of the ship.

**Comment by urushira** [comment]
Wasnt attached to land. The ship is iron. The error happens if I leave the ship while still in command of it. The blocks disappear while moving or changing direction. 
The "line" was added after the breakup to connect what was left together so I could move it to a build site. 
I dont know if its related but I suspect it is. I built a new airship today. 57 blocks in length with 3000 blocks. During flight, the server crashed. Upon login, the new ship was completely gone. i cant find it anywhere. Like the pierces on the zepplin, the entire ship just vanished.

**Comment by msummers123** [comment]
Yeah, the "too big to check borders" is simply because it's larger than 10k. I put that in because previously it would lag the server whenever the pilot left big ships. I wish I knew what to tell you other than I haven't seen this before and don't think it's very common. If you could get server logs during the error, that might help. Perhaps there is an error trace we can look through.

The only other thing I can think of is it could be caused by the somewhat unusual way blocks link together in movecraft. Vertically diagonally adjacent blocks will link to the ship, whereas horizontally diagonally adjacent blocks will not. If this was the issue, you should be able to see the old section that dropped off the ship behind the ship.

**Comment by urushira** [comment]
I've been watching this, more in a tiny bit of fear of losing work than anything else, but after having the smaller 3000 block ship deleted right from under me during a server crash, Ive begun to suspect that it has to do with travel speed. I was able to see the vertical slicing as the ship moved and that kind of queued me in.. The zepplin at 100K blocks was moving at only 7.75, while the smaller ship that got erased, was moving at 9.75. Top speed is 10.0 as you know, so there certainly seems to be a size/speed ratio that I have exceeded ( i.e. the larger the ship, the slower it needs to travel to be safe ). If this is a possibility, then I would love a bit of confirmation. The zepplin has been repaired at its final location and wont be moved again, as its part of an environment now, but other ships like the 8K sized replacement for the deleted ship will come along if for no other reason than being luxury busses for premium users to cross the nearly 15Kx18K size of our world.. I would like to know that paying players wont just be unceremoniously dropped from 200 blocks up after a server crash.. I am willing to guinea pig this and feed you result data if you could look into it. I dont know of many other people that  build on the scale that I build ( i build 1:1 scale ) and consider myself to be in the extreme range of builders.. 
![2015-12-23_02 26 56](https://cloud.githubusercontent.com/assets/16305786/12007273/c2a3c224-abb1-11e5-8884-1f93fbd6a233.png)
![2015-12-24_06 27 37](https://cloud.githubusercontent.com/assets/16305786/12007274/c2a81414-abb1-11e5-806a-d8c0bbeb830e.png)

**Comment by urushira** [comment]
heres a couple going backwards in dates from the 23rd of  dec.  


[2015-12-23-3.log.txt](https://github.com/msummers123/Movecraft-3/files/72483/2015-12-23-3.log.txt)
[2015-12-20-1.log.txt](https://github.com/msummers123/Movecraft-3/files/72484/2015-12-20-1.log.txt)
Theres not much out of the ordinary in the second and earlier log entry, but the first which happened on the day the little ship was deleted has an abnormality that ive seen a few times..
